packages/svelte/src/lib/utils.svelte.ts

```diff
@@ -0,0 +1,64 @@
+import { hasContext, getContext, setContext, untrack } from 'svelte';
+import { SvelteMap } from 'svelte/reactivity';
+
+export function createContext<T>(name: string) {
+  const key = Symbol(name);
+  return {
+    hasContext: () => {
+      try {
```

Comment:
this block is a little cryptic, probably deserves a brief comment. might be worth adding an API to Svelte for this at some point

```suggestion
      // At the time of writing there's no way to determine if we're
      // currently initializing a component without a try-catch
      try {
```